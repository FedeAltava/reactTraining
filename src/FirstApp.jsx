import PropTypes from 'prop-types'
export const FirstApp = ({title}) => {
  return (
    <>
        <h1>{title}</h1>
        <p>el fede</p>
    </>
  )
}

FirstApp.PropTypes = {
  title:PropTypes.string
}
