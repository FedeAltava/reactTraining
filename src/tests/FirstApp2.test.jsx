
import { describe, test, expect} from "@jest/globals";
import { render, screen} from "@testing-library/react";
import { FirstApp } from "../FirstApp";

describe('FirstApp', ()=>{
    const title = 'Manuuueeeeeeee'
    test('sholud match with snapchot',()=>{
        const { container } = render(<FirstApp title={title}/>)
        expect(container).toMatchSnapshot();
    });

    test('sholud render the title',()=>{
        render(<FirstApp title={title}/>)
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('sholud render the title',()=>{
        render(<FirstApp title={title}/>)
        expect(screen.getByRole('heading',{level:1})).toBeTruthy();
    });
});
