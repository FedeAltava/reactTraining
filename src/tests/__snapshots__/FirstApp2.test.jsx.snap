// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstApp sholud match with snapchot 1`] = `
<div>
  <h1>
    Manuuueeeeeeee
  </h1>
  <h2>
    Fede Altava
  </h2>
  <h3>
    No hay subtitulo
  </h3>
</div>
`;
